### ISSUE TITLE
Create demodulation matrices for new wave regions

### ISSUE THREAD

**Issue by mgalloy** [NCAR/ucomp-pipeline#249]
Create a demodulation matrix for each of the new wave regions from data observed in late November 2022. Use `UCoMP_Pol_Cal.pro` to create a demodulation matrix for each day with polarimetric data after 20221026. Then use `UCoMP_Pol_Cal_Temperature.pro` to combine days results to give a demodulation matrix with a temperature dependence.

**Comment by mgalloy** [comment]
I've gotten Steve's code running and produced D matrices for 20221118 and 20221119.

- [20221118 plots](https://github.com/NCAR/ucomp-pipeline/files/14398297/20221118.pdf)
- [20221119 plots](https://github.com/NCAR/ucomp-pipeline/files/14398298/20221119.pdf)

**Comment by mgalloy** [comment]
I ran 20221026 as a test date to compare Steve's results to mine to make sure I am getting the same results as Steve on a date where I had Steve's results. The demodulation matrices are all exactly the same, but the temperatures are wrong, mine are showing up as 0.0.

For example, for me:

```
Wavelength:      637.400
Unfiltered Temperature:      0.00000
```

and for Steve:

```
Wavelength:      637.400
Unfiltered Temperature:      32.7730
```

This will affect creating the temperature dependent fit in the next step.

**Comment by mgalloy** [comment]
I have fixed the issue with the temperatures. Mine now reads the same as Steve's:

```
Wavelength:      637.400
Unfiltered Temperature:      32.7730
```

**Comment by mgalloy** [comment]
The `UCoMP_Pol_Cal_Temperature` fit is failing with:

```
           2 sav files
           4 total calibrations at       637.400
           1 cals rejected due to bad chisq
       0 points identified as outliers
... large array of all zeros omitted ...
           1 cals rejected due to recurrent bad points
% Compiled module: POLY_FIT.
% POLY_FIT: Singular matrix detected.
```

I did check to see if the temperatures were different and they were:

```
      33.1440      33.1550      33.0770      33.0790
```

**Comment by mgalloy** [comment]
I don't think we have enough reading to fit the temperature. Using `UCoMP_Pol_Cal_Temperature` I had to include the August-November calibrations to get a 1074 nm temperature dependence, about 48 1074 nm calibrations — the 35 calibrations from September-November was not enough. We only have 2-4 calibrations for the new wave regions.

**Comment by bberkeyU** [comment]
Looking at the de-modulation coefficients across the wave regions.  It seems like there is a fairly smooth quadratic relationship between the wavelength and coefficient.  So we should be able to safely project to new wavelengths. 


https://github.com/NCAR/ucomp-studies/blob/master/demodulation/de-modcheck.ipynb

**Comment by mgalloy** [comment]
@bberkeyU How much variation of temperature do you expect now? If there is no (or very little) variation, we can make a constant demodulation matrix even from the few readings we have.

**Comment by bberkeyU** [comment]
@mgalloy I pulled the tu_mod values from the database for Dec 2021->Nov 2022.   The results were not exactly what I expected.  Somewhere in April 2022, it looks like the temperature started slowly rising, moving toward 32.5C.  We also have some big excursions over the course of a couple of days.

![image](https://github.com/NCAR/ucomp-pipeline/assets/22605316/66eb4310-6b34-4500-b63d-3df6c2920659)

**Comment by detoma** [comment]
We need to test if the recent de-modulation coefficients taken after the closure are consistent with the old ones.

**Comment by mgalloy** [comment]
The 20220409 polarimetric calibration data, especially for 1074 nm, is not in line with the coefficient fits (see #265). Waiting for more polarimetric calibration data when we reopen to create these demodulation matrices.

**Comment by detoma** [comment]
The 20220409 polarimetric calibration coefficient for the old wavelengths, especially for 1074 and 1079 nm are not in agreement with the old demodulation coefficients (see closed ticket https://github.com/NCAR/ucomp-pipeline/issues/265). In some cases they differ by 10% or more. Plots were produced by Ben to compare the new coefficients with the old ones. The plots can be found in this notebook:

https://github.com/NCAR/ucomp-studies/blob/master/demodulation/Plot%20UCoMP%20demodulation%20vs%20temp%20with%20new%202024%20data.ipynb

We do not have an explanation for why there is such difference and it is hard to draw conclusions with only one data point. We need to take more polarimetric data when the observatory reopens.
